python/plugins/phidata/composio_phidata/toolset.py

```diff
@@ -42,37 +43,48 @@ def _wrap_tool(
         # Create a new Toolkit instance
         toolkit = Toolkit(name=name)
 
-        @validate_call
-        def function(**kwargs: t.Any) -> str:
-            """Composio tool wrapped as Phidata `Function`.
+        # Get function parameters from schema
+        params = shared.get_signature_format_from_schema_params(parameters)
+
+        # Create function signature and annotations
+        sig = Signature(parameters=params)
+        annotations = {p.name: p.annotation for p in params}
+        annotations["return"] = str  # Add return type annotation
 
-            Args:
-                **kwargs: Function parameters based on the schema
+        @validate_call
+        def function_template(*args, **kwargs) -> str:
+            # Bind the arguments to the signature
+            bound_args = sig.bind(*args, **kwargs)
+            bound_args.apply_defaults()
 
-            Returns:
-                str: JSON string containing the function execution result
-            """
             return json.dumps(
                 self.execute_action(
                     action=Action(value=name),
-                    params=kwargs,
+                    params=bound_args.arguments,
                     entity_id=entity_id or self.entity_id,
                     _check_requested_actions=True,
                 )
             )
 
-        # Set function docstring from schema
-        param_docs = []
+        # Apply the signature and annotations to the function
+        function_template.__signature__ = sig
+        function_template.__annotations__ = annotations
+
+        # Format docstring in Phidata standard format
+        docstring_parts = [description, "\nArgs:"]
         if "properties" in parameters:
             for param_name, param_info in parameters["properties"].items():
                 param_desc = param_info.get("description", "No description available")
                 param_type = param_info.get("type", "any")
-                param_docs.append(f":param {param_name}: {param_desc} ({param_type})")
+                docstring_parts.append(f"    {param_name} ({param_type}): {param_desc}")
 
-        function.__doc__ = f"{description}\n\n" + "\n".join(param_docs)
+        docstring_parts.append(
+            "\nReturns:\n    str: JSON string containing the function execution result"
+        )
+        function_template.__doc__ = "\n".join(docstring_parts)
 
         # Register the function with the toolkit
-        toolkit.register(function)
+        toolkit.register(function_template)
```

Comment:
The code in lines 46-87 attempts to create a function with a dynamic signature using `inspect.Signature`.  However, it doesn't handle cases where the schema might be missing or malformed, leading to `KeyError` exceptions or incorrect function signatures.  This could cause runtime errors when calling the generated tools. The fix involves adding robust error handling (e.g., `try-except` blocks) to gracefully handle missing or invalid schema parameters and provide informative error messages.